dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.2.6.2/src/org/apache/cxf/transport/http/Headers.java

```diff
@@ -206,16 +206,18 @@ public void setProxyAuthorization(String authorization) {
      */
     public static Map<String, List<String>> getSetProtocolHeaders(final Message message) {
         Map<String, List<String>> headers =
-            CastUtils.cast((Map<?, ?>)message.get(Message.PROTOCOL_HEADERS));        
+            CastUtils.cast((Map<?, ?>)message.get(Message.PROTOCOL_HEADERS));
         if (null == headers) {
-            headers = new TreeMap<String, List<String>>(String.CASE_INSENSITIVE_ORDER);
+            headers = new TreeMap<String, List<String>>(String.CASE_INSENSITIVE_ORDER); 
+            message.put(Message.PROTOCOL_HEADERS, headers);
+            return headers;
         } else if (headers instanceof HashMap) {
-            Map<String, List<String>> headers2 
+            Map<String, List<String>> headers2
                 = new TreeMap<String, List<String>>(String.CASE_INSENSITIVE_ORDER);
             headers2.putAll(headers);
-            headers = headers2;
+            message.put(Message.PROTOCOL_HEADERS, headers2);
+            return headers2;
```

Comment:
I am guessing that part of the goal of the previous version of this code was to only have one return statement which is why it is coded that way.  It is a common coding practice to have one return statement in a method.

Reply:
Also all these changes are missing the Liberty change begin / end comments.